clients/da-vinci-client/src/test/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTaskTest.java

```diff
@@ -0,0 +1,183 @@
+package com.linkedin.davinci.kafka.consumer;
+
+import static com.linkedin.venice.writer.VeniceWriter.ENABLE_CHUNKING;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.linkedin.davinci.stats.AggVersionedDIVStats;
+import com.linkedin.davinci.stats.AggVersionedIngestionStats;
+import com.linkedin.davinci.stats.HostLevelIngestionStats;
+import com.linkedin.davinci.storage.chunking.ChunkingUtils;
+import com.linkedin.venice.kafka.protocol.KafkaMessageEnvelope;
+import com.linkedin.venice.kafka.protocol.Put;
+import com.linkedin.venice.kafka.protocol.enums.MessageType;
+import com.linkedin.venice.message.KafkaKey;
+import com.linkedin.venice.partitioner.DefaultVenicePartitioner;
+import com.linkedin.venice.utils.ByteUtils;
+import com.linkedin.venice.utils.SystemTime;
+import com.linkedin.venice.utils.VeniceProperties;
+import com.linkedin.venice.utils.lazy.Lazy;
+import com.linkedin.venice.writer.KafkaProducerWrapper;
+import com.linkedin.venice.writer.VeniceWriter;
+import com.linkedin.venice.writer.VeniceWriterOptions;
+import java.nio.ByteBuffer;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.mockito.ArgumentCaptor;
+import org.mockito.stubbing.Answer;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class ActiveActiveStoreIngestionTaskTest {
```

Comment:
Do you think we can reuse `StoreIngestionTaskTest` for AA, e.g. extend it?

The reason I am asking is we have done heavy mocking in `StoreIngestionTaskTest` already so it can save your time to mock them again. Besides, we don't need to make these methods, i.e. `getVersionedDIVStats``getVersionIngestionStats`, `getHostLevelIngestionStats`, and `getKafkaVersionTopic`, public only for testing purposes since it's been taken cared by the `StoreIngestionTaskTest`

Reply:
Actually I think StoreIngestionTaskTest is not very easy to be extended for this particular test case. The traditional StoreIngestionTaskTest does create a full SIT with mocked parameters and run ingestion task kind of e2e and verify properties. While in this test case, all I want to do is to very a behavior inside AASIT's single method. I created mock for this as I need to mock the producer to capture everything and verify every record it produces to VT.
If in the future we need to verify more AASIT's behavior, actually we can reuse these mocking to make it shared between different tests in this class.
Also, getXXX methods are needed as I am creating mocked SIT to test a basic method. It is actually not public, it is protected.